### ISSUE TITLE
Port technical documentation from shapechange.net to GitHub

### ISSUE THREAD

**Issue by jechterhoff** [ShapeChange/ShapeChange#281]
The technical documentation of ShapeChange should be hosted in the ShapeChange GitHub repository. 

Benefits of doing so:

* GitHub pages can be used to host and publish the documentation (rather than having to use a separate service).
* The documentation that applies to each release of ShapeChange is clearly documented in the source files of that release.
* Any enhancement or bugfix can directly update the relevant parts of the documentation as well. That applies to direct commits as well as Pull Requests from contributors (who could not directly update the documentation before, since the separate service to host the documentation required having an account).

The format of the technical documentation should be asciidoc. It can then be transformed to html pages using the asciidoctor maven plugin.

The [developer documentation](https://github.com/ShapeChange/ShapeChange/wiki/Developer-Documentation) of ShapeChange should be extended with information on how to use the asciidoctor maven plugin (e.g. using the auto-refresh goal while working on a documentation update, as well as during the release process), potentially links to relevant ascii(doctor) pages, as well as a cheat sheet with useful asciidoc syntax (but without the intent to recreate other cheat sheets that can be found on the web, or to repeat large parts of the asciidoc documentation).

**Comment by heidivanparys** [comment]
What is the status of this issue?

**Comment by jechterhoff** [comment]
@heidivanparys We have ported the documentation to asciidoc (see https://github.com/ShapeChange/ShapeChange/tree/master/src/docs/asciidoc). I then tried to set up a static site generator, to generate the website and hosting it on github pages. However, that did not work so well, the whole affair got more complex than I expected. So the issue is still open. I've set the milestone for the next ShapeChange version.
